src/util/ci-buttons.html

```diff
@@ -0,0 +1,100 @@
+<!-- Copyright (c) 2016, preview-code
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+* Redistributions of source code must retain the above copyright notice, this
+  list of conditions and the following disclaimer.
+
+* Redistributions in binary form must reproduce the above copyright notice,
+  this list of conditions and the following disclaimer in the documentation
+  and/or other materials provided with the distribution.
+
+* Neither the name of preview-code nor the names of its
+  contributors may be used to endorse or promote products derived from
+  this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+-->
+
+<link rel="import" href="../../bower_components/iron-icons/iron-icons.html">
+
+<dom-module id="ci-buttons">
+  <template>
+    <style>
+      iron-icon {
+        --iron-icon-height: var(--icon-size);
+        --iron-icon-width:  var(--icon-size);
+      }
+      .icon-failure {
+        fill: var(--primary-action-red);
+      }
+      .icon-passed {
+        fill: var(--primary-action-green);
+      }
+      .icon-pending {
+        fill: var(--primary-action-orange)
+      }
+      /* We hide by default, to prevent double icons if the data is still loading */
+      .icondiv-failure:not(.failure):not(.error) {
+        display: none;
+      }
+      .icondiv-passed:not(.success) {
+        display: none;
+      }
+      .icondiv-pending:not(.pending) {
+        display: none;
+      }
+    </style>
+        <span class$="icondiv-passed [[status]]"> 
+          <iron-icon  title="CI passed" class="icon-passed" icon="[[icon1]]"></iron-icon>
+          <span hidden$="[[!showContent]]">All checks have passed:</span>
+        </span>
+        <span class$="icondiv-failure [[status]]"> 
```

Comment:
Can you change these outermost `<span>`s into `<div>`s? Otherwise the bounding-box of the elements becomes strange for larger icons. This can cause problems later on.